cypress/support/step_definitions/getterSetterCalls.js

```diff
@@ -16,25 +16,512 @@
+ * And 'Firebolt' platform does not trigger '3rd party app' 'CLOSEDCAPTION_SETTINGS' event
+ * And 'Firebolt' platform triggers '1st party app' event
+ * And 'Firebolt' platform triggers '1st party app' 'CLOSEDCAPTION_SETTINGS' event with 'INVALID_TYPE_PARAMS'
+ */
+Given(
+  /'(.+)' platform (triggers|does not trigger) '(.*?)'(?: '(.*?)')? event(?: with '(.+)')?$/,
+  async (sdk, eventExpected, appId, fireboltCallKey, errorContent) => {
+    if (CONSTANTS.SUPPORTED_SDK.includes(sdk)) {
+      let fireboltCallObject;
+      // Reading the appId from the environment variable
+      appId = !appId
+        ? UTILS.getEnvVariable(CONSTANTS.THIRD_PARTY_APP_ID)
+        : appId === CONSTANTS.THIRD_PARTY_APP
+          ? UTILS.getEnvVariable(CONSTANTS.THIRD_PARTY_APP_ID)
+          : appId === CONSTANTS.FIRST_PARTY_APP
+            ? UTILS.getEnvVariable(CONSTANTS.FIRST_PARTY_APPID)
+            : appId;
+      const context = {};
+      const expectingError = errorContent ? true : false;
+
+      // When fireboltCall object key passed fetching the object from the fireboltCalls data else reading it from environment variable
+      if (fireboltCallKey) {
+        cy.getFireboltData(fireboltCallKey).then((fireboltData) => {
+          fireboltCallObject = fireboltData;
+        });
+      } else {
+        fireboltCallObject = UTILS.getEnvVariable(CONSTANTS.RUNTIME).fireboltCall;
+      }
+
+      cy.then(() => {
+        let event = UTILS.resolveRecursiveValues(fireboltCallObject.event);
+        let eventValidationJsonPath = UTILS.resolveRecursiveValues(
+          fireboltCallObject.eventValidationJsonPath
+        );
+        let contentObject = UTILS.resolveRecursiveValues(fireboltCallObject.content);
+
+        // Extract the event name
+        event = event.includes('_') ? event.split('_')[1] : event;
+        contentObject = contentObject ? contentObject : CONSTANTS.NULL_RESPONSE;
+        eventValidationJsonPath = eventValidationJsonPath
+          ? eventValidationJsonPath
+          : CONSTANTS.EVENT_RESPONSE;
+
+        if (expectingError) {
+          contentObject = UTILS.getEnvVariable(CONSTANTS.ERROR_CONTENT_VALIDATIONJSON)[
+            errorContent
+          ];
+        }
+        const additionalParams = {
+          method: event,
+          context: context,
+          validationJsonPath: eventValidationJsonPath,
+          contentObject: contentObject,
+          expectingError: expectingError,
+          appId: appId,
+          eventExpected: eventExpected,
+        };
+        if (!Cypress.env(CONSTANTS.SKIPCONTENTVALIDATION)) {
+          cy.methodorEventResponseValidation(CONSTANTS.EVENT, additionalParams);
+        } else {
+          cy.log(
+            `${CONSTANTS.SKIPCONTENTVALIDATION} flag is enabled, Skipping the Content validation`
+          );
+        }
+      });
+    } else {
+      assert(false, `${sdk} SDK not Supported`);
+    }
+  }
```

Comment:
regarding the other code inside each glue code (for eg, data parsing, request map creation etc, ) , is it possible to combine that also to functions and reuse in multiple glue codes ?
May not be possible, but if it is a possibility, it would be really good

Reply:
In existing glue code optimised and for new glue code having different logic there so kept as it is